### ISSUE TITLE
Does the default training script for step 1 sft phase, calculate the loss for the prompt part?

### ISSUE THREAD

**Issue by chunhuizhang** [huggingface/open-r1#648]
the default sft training scripts:

accelerate launch --config_file=recipes/accelerate_configs/zero3.yaml src/open_r1/sft.py \
    --model_name_or_path Qwen/Qwen2.5-1.5B-Instruct \
    --dataset_name open-r1/OpenR1-Math-220k \
    --learning_rate 5.0e-5 \
    --num_train_epochs 1 \
    --max_seq_length 16384 \
    --per_device_train_batch_size 16 \
    --gradient_checkpointing \
    --bf16 \
    --use_liger_kernel \
    --output_dir data/Qwen2.5-1.5B-Open-R1-Distill

the completion_only_loss will be set false (self.completion_only_loss = "prompt" in first_example)

the OpenR1-Math-220k does not has the 'prompt' column ?

**Comment by lewtun** [comment]
Hi @chunhuizhang indeed we do not mask the prompt from the loss calculation, but I have not seen it make a huge difference in my experiments. If you have evidence to the contrary, I would be happy to hear it!

**Comment by arahul14** [comment]
You can first transform your data into language modeling format, then convert it into prompt completion format and then apply_chat_template function fro trl (not tokenizer). Data processed in this format will automotically be trained on completion only loss.
```python
    sample = {
            "prompt": [msg_dict for msg_dict in example['messages'] if msg_dict['role']=='user'],
            "completion": [msg_dict for msg_dict in example['messages'] if msg_dict['role']=='assistant']
            
            }
    sample = apply_chat_template(sample, tokenizer)
```

**Comment by deeplearning-LI** [comment]
> You can first transform your data into language modeling format, then convert it into prompt completion format and then apply_chat_template function fro trl (not tokenizer). Data processed in this format will automotically be trained on completion only loss.
> 
>     sample = {
>             "prompt": [msg_dict for msg_dict in example['messages'] if msg_dict['role']=='user'],
>             "completion": [msg_dict for msg_dict in example['messages'] if msg_dict['role']=='assistant']
>             
>             }
>     sample = apply_chat_template(sample, tokenizer)

thanks

**Comment by arahul14** [comment]
If you want to convert the data into chatML format, then you need to apply_chat_template otherwise it will work as it is.
But remember that it will not be in conversational format which is not suitable for chatbot applications.

**Comment by arahul14** [comment]
That's correct, but here the use-case was to mask user messages in conversational format. That feature is not supported in trl. It will mask user messages only in the case of prompt completion dataset format. 
We have still many use-cases which are not supported yet like what happens in multi-turn conversation. How can we mask all the user messages in multi-turn conversations and train only on assistant messages. These features are still not supported in trl.
You can find all these features in axolotl
